libraries/kotlin.test/common/src/main/kotlin/kotlin/test/Utils.kt

```diff
@@ -5,8 +5,52 @@
 
 package kotlin.test
 
+import kotlin.math.abs
+
 internal fun messagePrefix(message: String?) = if (message == null) "" else "$message. "
 internal expect fun lookupAsserter(): Asserter
 
 @PublishedApi // required to get stable name as it's called from box tests
-internal fun overrideAsserter(value: Asserter?): Asserter? = _asserter.also { _asserter = value }
\ No newline at end of file
+internal fun overrideAsserter(value: Asserter?): Asserter? = _asserter.also { _asserter = value }
+
+
+private fun checkAbsoluteTolerance(absoluteTolerance: Double, message: String?) {
+    if (absoluteTolerance < 0.0) {
+        fail(messagePrefix(message) + "Illegal negative absolute tolerance <$absoluteTolerance>.")
```

Comment:
Invalid `absoluteTolerance` value isn't a test assertion failure, so it should throw `IllegalArgumentException`. Including `messagePrefix` is also questionable, since it relates to the assertion.

Reply:
done